apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/Adaptable.kt

```diff
@@ -0,0 +1,7 @@
+package com.apollographql.apollo.api
+
+import com.apollographql.apollo.api.internal.ResponseAdapter
+
+interface Adaptable<T> {
+  fun adapter(): ResponseAdapter<T>
```

Comment:
A new interface for classes that can `Adapt` others. I'm not 100% convinced on the naming but there is some potential to uniformize a lot of the code in `RealApolloStore` for operation/fragments